show/main.py

```diff
@@ -1133,6 +1120,215 @@ def ntp(verbose):
+    snmp = db.cfgdb.get_table('SNMP')
+    snmp_header = ["Location"]
+    snmp_body = []
+    if json_output:
+        try:
+            if snmp['LOCATION']:
+                click.echo(snmp['LOCATION'])
+        except KeyError:
+            click.echo("SNMP Location not set")
+    else:
+        try:
+            if snmp['LOCATION']:
+                snmp_location = [snmp['LOCATION']['Location']]
+                snmp_body.append(snmp_location)
+        except KeyError:
+            snmp['LOCATION'] = ''
+        click.echo(tabulate(snmp_body, snmp_header))
+
+
+# '("show runningconfiguration snmp user")
+@snmp.command('user')
+@click.option('--json', 'json_output', required=False, is_flag=True, type=click.BOOL, 
+              help="Display the output in JSON format")
+@clicommon.pass_db
+def users(db, json_output):
+    """show SNMP running configuration user"""
+    snmp_users_keys = db.cfgdb.get_table('SNMP_USER')
+    snmp_users = snmp_users_keys.keys()
+    snmp_user_header = ['User', "Permission Type", "Type", "Auth Type", "Auth Password", "Encryption Type", 
+                        "Encryption Password"]
+    snmp_user_body = []
+    if json_output:
+        try:
+            if snmp_users_keys:
+                click.echo(snmp_users_keys)
+        except KeyError:
+            click.echo("SNMP User not set")
+    else:
+        for line in snmp_users:
+            snmp_user = line
+            try:
+                snmp_user_permissions_type = snmp_users_keys[line]['SNMP_USER_PERMISSION']
+            except KeyError:
+                snmp_user_permissions_type = 'Null'
+            try:
+                snmp_user_auth_type = snmp_users_keys[line]['SNMP_USER_AUTH_TYPE']
+            except KeyError:
+                snmp_user_auth_type = 'Null'
+            try:
+                snmp_user_auth_password = snmp_users_keys[line]['SNMP_USER_AUTH_PASSWORD']
+            except KeyError:
+                snmp_user_auth_password = 'Null'
+            try:
+                snmp_user_encryption_type = snmp_users_keys[line]['SNMP_USER_ENCRYPTION_TYPE']
+            except KeyError:
+                snmp_user_encryption_type = 'Null'
+            try:
+                snmp_user_encryption_password = snmp_users_keys[line]['SNMP_USER_ENCRYPTION_PASSWORD']
+            except KeyError:
+                snmp_user_encryption_password = 'Null'
+            try:
+                snmp_user_type = snmp_users_keys[line]['SNMP_USER_TYPE']
+            except KeyError:
+                snmp_user_type = 'Null'
+            snmp_user_body.append([snmp_user, snmp_user_permissions_type, snmp_user_type, snmp_user_auth_type, 
+                                   snmp_user_auth_password, snmp_user_encryption_type, snmp_user_encryption_password])
+        click.echo(tabulate(natsorted(snmp_user_body), snmp_user_header))
+
+
+# ("show runningconfiguration snmp")
+@clicommon.pass_db
+def show_run_snmp_global(db, ctx):
```

Comment:
>global [](http://example.com/codeflow?start=18&length=6)

What is the meaning of `global`?

Reply:
It has both the "contact" and the "location" and it is the root "SNMP" for the redis-db.  
If you'd like it named different that's fine.  Just let me know what you'd want.  
I also named it like this because "show run snmp" will print all out these SNMP commands 
show run snmp contact
show run snmp location
show run snmp community
show run snmp user
So I called it "global" because it encompassed all of these commands.

Reply:
removed "Global" after looking at what you were saying.